packages/@webex/plugin-cc/src/services/core/WebSocket/WebSocketManager.ts

```diff
@@ -150,7 +156,7 @@ export class WebSocketManager extends EventTarget {
     this.isSocketClosed = true;
     this.keepaliveWorker.postMessage({type: 'terminate'});
     if (this.shouldReconnect) {
-      this.dispatchEvent(new Event('socketClose'));
+      this.emit('socketClose');
```

Comment:
As per Arun's advice, stuck to using event emitter and dispatching the events.
Added this all over and updated the event handling as well.